src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -746,7 +749,7 @@ gl_sframe drawing_classifier::perform_inference(data_iterator* data) const {
       pending_batches.pop();
 
       size_t num_images = batch.data_info.num_samples;
-      ASSERT_EQ(num_images * num_classes, batch.data_info.predictions.size());
+      ASSERT_EQ(batch_size * num_classes, batch.data_info.predictions.size());
```

Comment:
batch_size may not be real_batch_size, say you have 65 images, the batch size is 64, then you have 2 batches of 64 and 1.

Reply:
Well the point of going this is that there will be two batch of 64. The batch size would never change. The weights are to weigh the loss. The weights for the first batch would all ones and for the second batch will be [1, 0, 0 ...]. The predictions come from the softmax would also be equal to the batch_size=64 but if you see the code after this, it will compute accuracy only till the num samples in the batch (which would be the real batch size)

Reply:
I see. What's the benefit of doing this? I'm curious, if the batch size is 64, and for the current batch, I only have 1 image, what you will do? Padding the rest 63 images?

Reply:
Will this padding affect the batch norm?

Reply:
There is no batch norm in drawing classifier.

Reply:
Very good observe!
I also wondering why in our code we fix the batch size,
for instance, in AC we are doing the same thing and there are batch norm layers in the model.
And we might also change the model structure to something else for DC in the future (don't know this would happen tho haha), which might also include some cross-samples operation :).
But for now in the interest of consistent it makes sense to fix the batch size I guess haha.

Reply:
@shreyajain17 I mean for other toolkits with batch norms. I think this skewed batch norm will affect the final results.